### PR TITLE
enhancement/15579-custom-indicator-on-event 

### PR DESCRIPTION
**Pull request by karolkolodziej**
Enhancement #15579, added documentation with an example on how to calculate custom indicators on events.

### CODE REVIEW

**Review by pawelfus — CHANGES_REQUESTED** [review]
We need to put this on hold, because of https://github.com/highcharts/highcharts/issues/16397 - it seems a minor refactoring is necessary: indicator should be bound to both `updatedData` and `afterSetExtremes` events, not only one of them

File: docs/stock/custom-technical-indicators.md [hunk 1]
```diff
@@ -287,3 +287,30 @@ A live demo of the above multiline indicator can be found [here](https://jsfiddl
 
 _For more detailed samples and documentation check the [API](https://api.highcharts.com/highstock/series.trendline)._
 
+### 4. Indicators calculated on events.
+
+The idea behind creating a custom indicator that is calculated after some chart evet (for example `afterSetExtremes`) is similar to the ones previously mentioned with a few additions.
```
Comment (pawellysy) [inline docs/stock/custom-technical-indicators.md:292]: `evet` -> `event`

File: docs/stock/custom-technical-indicators.md [hunk 2]
```diff
@@ -287,3 +287,30 @@ A live demo of the above multiline indicator can be found [here](https://jsfiddl
...
+
+The most important object to add while creating the indicators is `bindTo` where inside are specified two properties:
+*   `series` - whether the indicator should be calculated together with the linked main series,
+*   `eventName` - on which event the indicator should be calculated,
+
+The indicator code should look like:
```
Comment (pawellysy) [inline docs/stock/custom-technical-indicators.md:298]: `Example configuration should look like this`

**Review by pawellysy — CHANGES_REQUESTED** [review]
Looks good! Some minor polishing :)

**Review by raf18seb — APPROVED** [review]
Thanks!

File: docs/stock/custom-technical-indicators.md [hunk 3]
```diff
@@ -287,3 +287,30 @@ A live demo of the above multiline indicator can be found [here](https://jsfiddl
 
 _For more detailed samples and documentation check the [API](https://api.highcharts.com/highstock/series.trendline)._
 
+### 4. Indicators calculated on events.
+
+The idea behind creating a custom indicator that is calculated after some chart event (for example `afterSetExtremes`) is similar to the ones previously mentioned with a few additions.
+
+The most important object to add while creating the indicators is `calculateOn` where inside are specified two properties:
+*   `chart` - on which chart's event the indicator should be calculated, mainly you can choose from `init` (before the linked series is processed) and `render` (after processing the linked series).
+*   `xAxis` - on which xAxis' event the indicator should be recalculated,
```
Comment (pawelfus) [inline docs/stock/custom-technical-indicators.md:296]: Could you add an example? Something like this: 
> on which xAxis' event the indicator should be recalculated (e.g. `afterSetExtremes`)

**Review by pawelfus — CHANGES_REQUESTED** [review]
Just a minor request.

Note: This PR should not be merged until the new version release (because docs are more often updated, and we don't want to have docs that describe something that will be possible in the next version). Should we somehow label this PR, @TorsteinHonsi ?

**Review by pawellysy — APPROVED** [review]
I like what you got. Good job!

**Review by pawelfus — APPROVED** [review]
#16397 merged and released, we can do the same with this docs-related PR.
Thanks!
PS: Awaiting CI

**Comment by TorsteinHonsi** [comment]
So the _on hold_ comment in the first comment is no longer valid?
Also, merging in the master will probably fix the CI failures.

**Comment by pawelfus** [comment]
> So the on hold comment in the first comment is no longer valid?

Yes, that's right.

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**